primitives/linux-port/column.cpp

```diff
@@ -1185,17 +1331,22 @@ void scalarFiltering(
     const bool validMinMax,     // The flag to store min/max
     T emptyValue,               // Deduced empty value magic
     T nullValue,                // Deduced null value magic
-    T Min, T Max, const bool isNullValueMatches)
+    T Min, T Max, const bool isNullValueMatches,
+    const uint8_t* blockAux,
+    uint8_t emptyValueAux)
 {
   constexpr int WIDTH = sizeof(T);
   // Loop-local variables
   T curValue = 0;
   primitives::RIDType rid = 0;
   bool isEmpty = false;
 
+  auto nextColValuePtr = in->hasAuxCol ? nextColValue<T, WIDTH, true> : nextColValue<T, WIDTH, false>;
```

Comment:
What if you move this one level up so that scalarFiltering doesn't check in->hasAuxCol for every value? You did the similar thing for the vectorized processing path.

Reply:
My bad I should have told you this right away.

Reply:
This branching is outside the loop, so `in->hasAuxCol` is only checked per call, not for every value in the loop.

Reply:
You are right.